qiime/core/type/primitive.py

```diff
@@ -349,6 +349,24 @@ def encode(self, value):
 Float = _Float('Float')
 
 
+class _Bool(_Primitive):
+    _valid_predicates = {Arguments,}
+
+    def _is_element_(self, value):
+        return isinstance(value, bool)
+
+    def decode(self, string):
+        if string not in ('false', 'true'):
+            raise TypeError("%s is neither 'true' or 'false'")
+
+        return string == 'true'
+
+    def encode(self, value):
+        return str(value)
```

Comment:
`str(42)` returns `'42'`, which is the correct JSON representation of an integer. `str(True)` returns `'True'`, which is not a valid JSON boolean type. Does that make sense? `encode`/`decode` really need to be renamed `to_json`/`from_json`.